common/src/main/java/org/opensearch/ml/common/ToolMetadata.java

```diff
@@ -0,0 +1,88 @@
+package org.opensearch.ml.common;
+
+import lombok.Builder;
+import lombok.Getter;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+
+
+public class ToolMetadata implements ToXContentObject, Writeable {
+
+    public static final String TOOL_NAME_FIELD = "name";
+    public static final String TOOL_DESCRIPTION_FIELD = "description";
+
+    @Getter
+    private String name;
+    @Getter
+    private String description;
```

Comment:
do we need tool type here?

Reply:
Please leave a comment in this [issue](https://github.com/opensearch-project/ml-commons/issues/1802) if you believe we need to add tool type here

Reply:
Not sure who will be the consumer of this API, tool name is not enough for register a agent, if it should used by ai flow.